### PR TITLE
Upgrade Cilium to v1.14

### PR DESCRIPTION
**Pull request by FedericoAntoniazzi**
Upgrade Cilium for Fury v1.27.

## Features
- Upgrade Cilium to `v1.14.3`
- Upgrade Hubble to `v0.12.1`
- Add preliminary actions to decrease CNI downtime on upgrade 

The documentation introduces the release notes for the next version (v0.15.0) and describes the upgrade procedure for Cilium using the preflight script.

## Roadmap
- [x] Upgrade Cilium
- [x] Upgrade Hubble
- [x] Update docs
- [x] Add CI e2e tests for Kubernetes v1.27

## Notes
- The upgrade has been tested on a 5 nodes cluster created with Kind
- CNI functionalities have been tested by the `cilium connectivity test` command

### CODE REVIEW

File: docs/releases/v1.15.0.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+#### Cilium upgrade
+Cilium suggested path expect a pre-flight check to be run before any upgrade.
+
+1. Create the resources for the check
+```bash
+kubectl create -f katalog/cilium/tasks/preflight.yaml
+```
+
+2. Ensure that the number of READY pods is the same number of Cilium pods running.
```
Comment (mimnix) [inline docs/releases/v1.15.0.md:42]: "Make sure that the number of READY pods is the same as the number of RUNNING Cilium pods."

File: katalog/cilium/MAINTENANCE.values.yaml
Comment (mimnix) [inline katalog/cilium/MAINTENANCE.values.yaml:1]: Besides the image version bump, are these all the "override with defaults" values removals, aren't they?

**Review by mimnix — CHANGES_REQUESTED** [review]
Just a minor change proposed in the docs and a check about the `MAINTENANCE.values.yaml` file.

File: docs/releases/v1.15.0.md (hunk 1 above)
Comment (FedericoAntoniazzi) [reply docs/releases/v1.15.0.md:42]:
```suggestion
2. Make sure that the number of READY pods is the same as the number of RUNNING Cilium pods.
```

File: katalog/cilium/MAINTENANCE.values.yaml
Comment (FedericoAntoniazzi) [reply katalog/cilium/MAINTENANCE.values.yaml:1]: yes, it is

**Review by nutellinoit — CHANGES_REQUESTED** [review]
The target branch should be develop

**Comment by FedericoAntoniazzi** [comment]
Changed the destination branch. will require some more testing to be sure no manifest has been broken solving merge conflicts.

**Review by nutellinoit — APPROVED** [review]
LGTM

**Pull request merged**